src/main/java/jimmy/mcgymmy/model/date/Date.java

```diff
@@ -0,0 +1,94 @@
+package jimmy.mcgymmy.model.date;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import jimmy.mcgymmy.commons.util.AppUtil;
+
+public class Date {
+    public static final String MESSAGE_CONSTRAINTS = "Date string should be in one of the following formats:\n"
+            + "yyyy-M-dd\n"
+            + "yyyy-M-d\n"
+            + "dd-MM-yyyy\n"
+            + "dd-M-yyyy\n"
+            + "d-M-yyyy\n"
+            + "yyyy/MM/dd\n"
+            + "dd/MM/yyyy\n"
+            + "dd/M/yyyy\n"
+            + "d/M/yyyy\n"
+            + "d MMM yyyy\n";
+
+    private static final String[] SUPPORTED_FORMATS = {
+        "yyyy-MM-dd",
+        "yyyy-M-dd",
+        "yyyy-M-d",
+        "dd-MM-yyyy",
+        "dd-M-yyyy",
+        "d-M-yyyy",
+        "yyyy/MM/dd",
+        "dd/MM/yyyy",
+        "dd/M/yyyy",
+        "d/M/yyyy",
+        "d MMM yyyy"
+    };
+
+    private static final String OUTPUT_FORMAT = "d MMM yyyy";
+
+    private LocalDate date;
+
+    /**
+     * Constructs a {@code Date}.
+     *
+     * @param date A valid date string.
+     */
+    public Date(String date) {
+        requireNonNull(date);
+        boolean canParse = false;
+        for (String format : SUPPORTED_FORMATS) {
+            try {
+                // check if the date is in this format
+                this.date = LocalDate.parse(date, DateTimeFormatter.ofPattern(format));
+                canParse = true;
+                break;
+            } catch (DateTimeParseException e) {
+                // exception is thrown means that cannot parse date in that format
+                // if cannot parse date in any format, throw IllegalArgumentException
+            }
```

Comment:
Even if the string is in the correct format, it is still not guaranteed that it can be parsed. For example, 31-31-2020. We still need the empty catch clause either way

Reply:
Hmm true. 
But I rmb hearing somewhere in 2103 that empty catch clause is bad thou.

Reply:
I remembered them said empty catch clause should be avoided, but if there is no other way to avoid it, we should comment why. I think putting a comment there is good enough

Reply:
Hmm ookie